Automattic/pocket-casts-android
# Use new onboarding flow throughout the app

| 📘 Project: #524  |
| --- |

## Description
This PR is bringing the screens from the onboarding flow to the rest of the app.

Sorry for the big PR. These flows were so interrelated that changing one could break any of the other ones so I thought it would be easier to just test them all at once as opposed to repeatedly re-testing them across multiple PRs.

## Testing Instructions

Test that the appropriate flows start from the appropriate location. With each flow, also verify that an appropriate "source" property is sent to Tracks when the Plus Upgrade screen is presented during the flow.

In parentheses, I note the flow tag that should be sent to tracks along with the events from that flow, i.e.: `setup_account_shown, Properties: {"flow":"plus_upsell"...`.

### A. Full onboarding flow

####  Locations

1.  First time the user opens the app after a new installation ("initial_onboarding" flow)
2.  Profile → Sign In ("logged_out" flow)

####  Flow

1.  Login or Sign Up
2.  If Login, exit flow, otherwise
4.  If created a new account: Recommendations (going back from here should exit the flow)
5.  Plus Upgrade
6.  Welcome (if the user upgraded, going back from here should exit the flow)

### B. Plus Upgrade flow

This is what we send if the user is indicating they want to upgrade (i.e., clicking an Upgrade button). Because the user is indicating they want upgrade already, we skip the Plus Upgrade\'s full screen view and immediately open the purchase bottom modal.

####  Locations

1. Profile tab → \"Signed in as\" box → Tap on upgrade/start-trial button
2. Profile tab → ⚙ → Plus Details → Upgrade Button

####  Signed In ("plus_account_upgrade" flow)

1.  Purchase modal
2.  After exiting the flow, the upgrade screen should be closed and the user should be returned to the settings screen

####  Signed Out ("plus_account_upgrade_needs_login" flow)

1.  Login or Sign up
2.  If created a new account: Recommendations
3.  Purchase bottom modal (dismissing the modal presents the full screen upsell page, then tapping Not Now or swiping back will exit the flow)
4.  If created a new account: Welcome. If logged into a previous account: exit the flow.
5.  After exiting the flow, the upgrade screen should be closed and the user should be returned to the settings screen

### C. Plus Upsell flow ("plus_upsell" flow)

This is what we send if the user has tapped on a Plus feature or
otherwise indicated they are interested in learning more about
upgrading.

####  Locations

1.  Profile tab → ⚙ → Appearance → Plus Theme
2.  Profile tab → ⚙ → Appearance → Plus Icon
3.  Profile tab → ⚙ → Appearance → Plus banner at bottom of screen
4.  Profile tab → Files → Add File → Image Placeholder
5.  Profile tab → Files → Add File → \"Add Custom Image\"
7.  Profile tab → Files → Add File → Tapping on a color with a 🔒
8.  Profile tab → Files → Add File → Plus banner at bottom of screen
9.  Profile tab → Files → Overflow menu → File settings → Tap on one of the Plus features
10.  Profile tab → Files → Overflow menu → File settings → Plus banner at bottom of screen
11. Profile tab → \"Signed in as\" box → Learn more about Pocket Casts
12. Profile tab → Plus banner at bottom of screen
13. Profile tab → \"Signed in as\" box → Start Free Trial / Upgrade button
14. Podcasts tab → Folders upgrade prompt

#### Signed In

1.  Plus Upgrade Full Screen
2.  Tapping button to upgrade presents purchase modal (alternatively, tapping Not Now or swiping back exits the flow)
3.  Welcome Screen

####  Signed Out

1.  Plus Upgrade Full Screen
2.  Tapping button to upgrade presents the Login or Sign Up flow
3.  If created a new account: Recommendations
4.  Plus Upgrade screen with the purchase modal automatically presented (tapping back will dismiss the purchase modal and present the full-screen plus upsell page)
5.  If created a new account: Welcome Screen (if you are in one of the File upsell flows, the Go to DIscover option will not be presented on this page. That was done for two reasons: (1) the user was in the middle of adding a file, so this avoids aborting that process and (2) navigating the user back between activities to the discover screen programmatically would be tricky and fragile to implement).

## Screenshots or Screencast 
This is what the full onboarding flow looks like. Note that the particular flow for a particular part of the app will likely be slightly different from this as described above in the Testing section.

https://user-images.githubusercontent.com/4656348/211891196-480d8de7-d2c9-43ba-8e42-b73884a2fdf5.mov


## Checklist
 
#### I have tested any UI changes...
